lib/client-assets.php

```diff
@@ -677,3 +677,22 @@ function gutenberg_extend_block_editor_preload_paths( $preload_paths, $post ) {
 	return $preload_paths;
 }
 add_filter( 'block_editor_preload_paths', 'gutenberg_extend_block_editor_preload_paths', 10, 2 );
+
+
+/**
+ * Extends block editor settings to include __experimentalPreferredBlockStyles.
+ * This option allows themes to set the default style applied in a block.
+ *
+ * @param array $settings Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function gutenberg_add_preferred_block_styles_setting( $settings ) {
+	$preferred_block_styles = current( (array) get_theme_support( 'editor-preferred-block-styles' ) );
```

Comment:
Any reason for everything else being experimental and not this: editor-preferred-block-styles

Reply:
The reason was that while I expected editor-preferred-block-styles to be the way themes set the default styles, I was not totally confident in the way we pass the settings to the block editor. We have the user preference and the theme preference, we pass a single default (join of both settings). Depending on the UI we may need to pass the user value and the theme value (e.g: if we want to show the default if the user resets its setting). As the block styles UI, did not receive much attention I thought it would be good to keep the options open.
But we can make editor-preferred-block-styles experimental, or stabilize the javascript settings given that an additional time passed and until now we did not face problems. Any thoughts on this?

Reply:
> The reason was that while I expected editor-preferred-block-styles to be the way themes set the default styles

Do you think the global styles work (theme.json) or the block-features proposal could change that too?